jbmc/src/java_bytecode/java_qualifiers.cpp

```diff
@@ -56,7 +56,7 @@ void java_qualifierst::write(typet &src) const
 qualifierst &java_qualifierst::operator+=(const qualifierst &other)
 {
   c_qualifierst::operator+=(other);
-  auto jq = dynamic_cast<const java_qualifierst *>(&other);
+  auto jq = static_cast<const java_qualifierst *>(&other);
```

Comment:
:no_entry_sign: 

This doesn't work with the following logic, because it checks for the failure of the dynamic cast. `static_cast` cannot fail in this way.